plugins/modules/ndo_ipsla_monitoring_policy.py

```diff
@@ -0,0 +1,398 @@
+    mso_template.validate_template("tenantPolicy")
+    object_description = "IPSLA Monitoring Policy"
+
+    path = "/tenantPolicyTemplate/template/ipslaMonitoringPolicies"
+    existing_ipsla_policies = mso_template.template.get("tenantPolicyTemplate", {}).get("template", {}).get("ipslaMonitoringPolicies", [])
+    if ipsla_monitoring_policy or ipsla_monitoring_policy_uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_ipsla_policies,
+            [KVPair("uuid", ipsla_monitoring_policy_uuid) if ipsla_monitoring_policy_uuid else KVPair("name", ipsla_monitoring_policy)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_ipsla_policies
+
+    if state == "present":
+        if match:
+            replace_data = {
+                "name": ipsla_monitoring_policy,
+                "description": description,
+                "slaType": sla_type,
+                "slaPort": destination_port,
+                "httpVersion": http_version,
+                "httpUri": http_uri,
+                "slaFrequency": sla_frequency,
+                "slaDetectMultiplier": detect_multiplier,
+                "reqDataSize": request_data_size,
+                "ipv4ToS": type_of_service,
+                "timeout": operation_timeout,
+                "threshold": threshold,
+                "ipv6TrfClass": ipv6_traffic_class,
+            }
+            update_path = "{0}/{1}".format(path, match.index)
+            append_update_ops_data(ops, match.details, update_path, replace_data)
+            mso.sanitize(match.details)
+
+        else:
+            payload = {"name": ipsla_monitoring_policy}
+
+            if description:
+                payload["description"] = description
+
+            if sla_type:
+                payload["slaType"] = sla_type
+
+            if destination_port:
+                payload["slaPort"] = destination_port
+
+            if http_version:
+                payload["httpVersion"] = http_version
+
+            if http_uri:
+                payload["httpUri"] = http_uri
+
+            if sla_frequency:
+                payload["slaFrequency"] = sla_frequency
+
+            if detect_multiplier:
+                payload["slaDetectMultiplier"] = detect_multiplier
+
+            if request_data_size:
+                payload["reqDataSize"] = request_data_size
+
+            if type_of_service:
+                payload["ipv4ToS"] = type_of_service
+
+            if operation_timeout:
+                payload["timeout"] = operation_timeout
+
+            if threshold:
+                payload["threshold"] = threshold
+
+            if ipv6_traffic_class:
+                payload["ipv6TrfClass"] = ipv6_traffic_class
+
+            ops.append(dict(op="add", path="{0}/-".format(path), value=payload))
+
+            mso.sanitize(payload)
```

Comment:
You can also move the `replace_data` outside the if statement and use it here. The `sanitize` function with `collate=True` will remove any unset keys and set `mso.sent` to the payload.

```suggestion
            mso.sanitize(replace_data, collate=True)
            ops.append(dict(op="add", path="{0}/-".format(path), value=mso.sent))
```